File: holbertonschool-higher_level_programming/0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
class Rectangle:
    def __init__(self, width=0, height=0):
        self.width = width
        self.height = height

    @property
    def width(self):
        return self.__width

    @width.setter
    def width(self, value):
        if type(value) is not (int):
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        return self.__height

    @height.setter
    def height(self, value):
        if type(value) is not (int):
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def __str__(self):
        new_string = ""
        for y in range(0, self.__height):
            for x in range(0, self.__width):
                new_string += "#"
            if y != (self.__height - 1):
                new_string += "\n"
        return (new_string)

    def area(self):
        if self.__width == 0 or self.__height == 0:
            return ("")
        return (self.__height * self.__width)

    def perimeter(self):
        if self.__width == 0 or self.__height == 0:
            return 0
        else:
            return ((self.__height * 2) + (self.__width * 2))

File: holbertonschool-higher_level_programming/0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node

class Rectangle {
  constructor (w, h) {
    if (w > 0 && h > 0) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    for (let outerLoop = 0; outerLoop < this.height; outerLoop++) {
      let string = '';
      for (let innerLoop = 0; innerLoop < this.width; innerLoop++) {
        string = string + 'X';
      }
      console.log(string);
    }
  }
}

module.exports = Rectangle;
